Add self-observability as a core engineering value (#3544)
open-telemetry/community: mission-vision-values.md

a performant application. High performance is a requirement for OpenTelemetry,
and unexpected interference effects in the host application are unacceptable.

### We value _self-observability_

> OpenTelemetry components should be exemplars of well-observed software. We

Reviewer: `Exemplars` is used correctly in this context. However `exemplar` is also part of the specification.
Could this be confusing?
Would an alternative wording improve this?
```suggestion
OpenTelemetry components should serve as examples of well-observed software. We
```

Author: I think the context is sufficient enough to avoid any confusion. "Exemplar" feels more apt as it implies best-in-class, not just mere examples.
Keeping as-is, let me know if this is blocking and I can re-assess.
